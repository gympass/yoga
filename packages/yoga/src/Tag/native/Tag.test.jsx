import React from 'react';
import { render } from '@testing-library/react-native';
import { Building } from '@gympass/yoga-icons';

import { ThemeProvider, Tag } from '../..';

describe('<Tag />', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag>default</Tag>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with custom icon and informative type', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag.Informative variant="success" icon={Building}>
          success with custom icon
        </Tag.Informative>
        <Tag.Informative variant="success" icon={Building} small>
          success with custom icon
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with without icon and informative type', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">
          success without icon
        </Tag.Informative>
        <Tag.Informative variant="success" small>
          success without icon
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with variant prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag variant="success">success</Tag>
        <Tag variant="informative">informative</Tag>
        <Tag variant="attention">attention</Tag>
        <Tag variant="attention" small>
          attention small
        </Tag>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with variant prop and informative type', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">success</Tag.Informative>
        <Tag.Informative variant="informative">informative</Tag.Informative>
        <Tag.Informative variant="attention">attention</Tag.Informative>
        <Tag.Informative variant="attention" small>
          attention small
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with variant and margin prop ', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag variant="success">success</Tag>
        <Tag mv="medium" variant="informative">
          informative
        </Tag>
        <Tag variant="attention">attention</Tag>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with variant prop and informative type with margin prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">success</Tag.Informative>
        <Tag.Informative mv="medium" variant="informative">
          informative
        </Tag.Informative>
        <Tag.Informative variant="attention">attention</Tag.Informative>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
