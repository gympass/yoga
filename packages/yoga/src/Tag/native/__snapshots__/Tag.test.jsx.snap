// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should match snapshot 1`] = `
<View
  small={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#9898A6",
        "borderRadius": 4,
        "borderWidth": 1,
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
        "width": "auto",
      },
    ]
  }
  variant=""
>
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "500",
        },
        Object {
          "color": "#6B6B78",
          "fontSize": 12,
          "lineHeight": 16,
        },
      ]
    }
  >
    default
  </Text>
</View>
`;

exports[`<Tag /> should match snapshot with custom icon and informative type 1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <RNSVGSvgView
        bbHeight={16}
        bbWidth={16}
        fill="#231B22"
        focusable={false}
        height={16}
        marginRight="xxxsmall"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Array [
              Object {
                "height": 16,
                "marginRight": 4,
                "width": 16,
              },
            ],
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        width={16}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4280490786,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success with custom icon
      </Text>
    </View>
  </View>,
  <View
    small={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 0,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <RNSVGSvgView
        bbHeight={12}
        bbWidth={12}
        fill="#231B22"
        focusable={false}
        height={12}
        marginRight="xxxsmall"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Array [
              Object {
                "height": 12,
                "marginRight": 4,
                "width": 12,
              },
            ],
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 12,
              "width": 12,
            },
          ]
        }
        width={12}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4280490786,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success with custom icon
      </Text>
    </View>
  </View>,
]
`;

exports[`<Tag /> should match snapshot with variant and margin prop  1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#1D856C",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#1D856C",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      success
    </Text>
  </View>,
  <View
    mv="medium"
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#5340C9",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="informative"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#5340C9",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      informative
    </Text>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#CA4808",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#CA4808",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      attention
    </Text>
  </View>,
]
`;

exports[`<Tag /> should match snapshot with variant prop 1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#1D856C",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#1D856C",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      success
    </Text>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#5340C9",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="informative"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#5340C9",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      informative
    </Text>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#CA4808",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#CA4808",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      attention
    </Text>
  </View>,
  <View
    small={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#CA4808",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 0,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "500",
          },
          Object {
            "color": "#CA4808",
            "fontSize": 12,
            "lineHeight": 16,
          },
        ]
      }
    >
      attention small
    </Text>
  </View>,
]
`;

exports[`<Tag /> should match snapshot with variant prop and informative type 1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success
      </Text>
    </View>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0DFFF",
          "borderColor": "#E0DFFF",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="informative"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        informative
      </Text>
    </View>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCD6C3",
          "borderColor": "#FCD6C3",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        attention
      </Text>
    </View>
  </View>,
  <View
    small={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCD6C3",
          "borderColor": "#FCD6C3",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 0,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        attention small
      </Text>
    </View>
  </View>,
]
`;

exports[`<Tag /> should match snapshot with variant prop and informative type with margin prop 1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success
      </Text>
    </View>
  </View>,
  <View
    mv="medium"
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0DFFF",
          "borderColor": "#E0DFFF",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="informative"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        informative
      </Text>
    </View>
  </View>,
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCD6C3",
          "borderColor": "#FCD6C3",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="attention"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        attention
      </Text>
    </View>
  </View>,
]
`;

exports[`<Tag /> should match snapshot with without icon and informative type 1`] = `
Array [
  <View
    small={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success without icon
      </Text>
    </View>
  </View>,
  <View
    small={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderColor": "#C1EEDB",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 0,
          "width": "auto",
        },
      ]
    }
    variant="success"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#231B22",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 16,
            },
          ]
        }
      >
        success without icon
      </Text>
    </View>
  </View>,
]
`;
